### PR TITLE
Add footer

### LINKED ISSUE #7: Create footer
- [x] Tagline is in place
- [x] Each icon links to proper platform
- [x] Correct colours are used
- [x] Footer is responsive

Desktop Footer:
Image

Mobile Footer:
Image

### PR DESCRIPTION
**Pull request by topanb11**
Resolves #7 
-  Added both desktop and mobile footer
-  Added social icons
-  Icons link to respective platforms 

Mobile View:
Screen Shot 2022-08-03 at 11 42 43 PM

Desktop View:
Screen Shot 2022-08-03 at 11 42 08 PM

### CODE REVIEW

File: components/Footer.jsx
Comment (harsohailB) [inline components/Footer.jsx]: I created a component called `SocialMediaIcons` in #14. We should re-use that here. Maybe wait until that merges and then pull the changes up into here to re-use.

https://github.com/Code-the-Change-YYC/code-the-change-yyc-site/blob/main/components/SocialMediaIcons.jsx

File: components/Footer.jsx
Comment (harsohailB) [inline components/Footer.jsx]: Seems like your prettier isn't working correctly. This should have also been pointed out by the ESLint. Can you add this to `.eslintrc.json`? This should flag wrongful styling in the check.

```josn
"rules": {
  "indent": ["error", 2]
}
```

File: pages/_app.js
Comment (harsohailB) [inline pages/_app.js]: Same styling fix needed here

File: components/Footer.jsx
Comment (harsohailB) [inline components/Footer.jsx]:
```suggestion
        <h4 className="text-[#C5C5C5] font-light mt-6">Copyright @ Code the Change YYC</h4>
```

File: components/Footer.jsx
Comment (harsohailB) [inline components/Footer.jsx]: I think you can use the `Link` component here from next instead of handling click yourself. No need to make a change tho because you can re-use the `SocialMediaIcons` component from the other PR #14

**Review by harsohailB — CHANGES_REQUESTED** [review]
Nice job! Just some styling and component re-use comments

File: yarn.lock
Comment (harsohailB) [inline yarn.lock]: Something messed up with the `yarn.lock`. The diff should not be that big (+7,809 −1,839). If you pull in master and just accept master's `yarn.lock` changes,  it should resolve I think.

File: components/Footer.jsx
Comment (topanb11) [reply components/Footer.jsx]: For the ```SocialMediaIcons``` component from #14, It doesn't change when switching to a mobile screen, should we open another branch to make it so the component is responsive?

File: components/Footer.jsx
Comment (topanb11) [reply components/Footer.jsx]: Added

File: pages/_app.js
Comment (topanb11) [reply pages/_app.js]: Done

File: components/Footer.jsx
Comment (topanb11) [reply components/Footer.jsx]: Just gonna wait for response to the comment from earlier before moving on with any changes

File: yarn.lock
Comment (topanb11) [reply yarn.lock]: Done, pulled in changes from master

File: components/Footer.jsx
Comment (harsohailB) [reply components/Footer.jsx]: You can pass in `props` in this PR to override the colors in this PR. Shouldn't be too big of a change

File: yarn.lock
Comment (harsohailB) [reply yarn.lock]: The diff still does not seem right. Maybe try `git checkout origin/master yarn.lock` then run `yarn`?

Comment (harsohailB) [reply yarn.lock]: Wait I don't see the commit from merging master. Can you push the pull from master

File: .prettierrc.json
Comment (harsohailB) [inline .prettierrc.json]: We want this in the `.eslintrc.json` so the github checks catch any wrong doing. The prettier already has the formatting with `"tabWidth": 2`

File: components/SocialMediaIcons.jsx
Comment (harsohailB) [inline components/SocialMediaIcons.jsx]: Not sure if this is the best way to do it as it screws up the other component (colors missing):

Screen Shot 2022-08-04 at 9 45 08 PM

I think what we want is something like this to override so we retain the classes original classes:

```js
const SocialMediaIcons = ({ classes }) => {
   ...
   <div className={`flex items-start pt-5 text-[#FFD2DC] space-x-6 ${classes}`}>
   ...
}
```

```js
<SocialMediaIcons classes="text-[#7055FD] md:text-[#FFFFFF]" />
```

**Review by harsohailB — CHANGES_REQUESTED** [review]
Much better. Gave some suggestions

File: components/SocialMediaIcons.jsx
Comment (topanb11) [reply components/SocialMediaIcons.jsx:8]: Wait that's weird because I can see the changes here. I'll take a look at it in a bit

Screen Shot 2022-08-04 at 10 51 01 PM
Screen Shot 2022-08-04 at 10 50 41 PM
Screen Shot 2022-08-04 at 10 50 50 PM

Comment (topanb11) [reply components/SocialMediaIcons.jsx]: Okay I fixed it but it's not passing through the linter now

File: .prettierrc.json
Comment (topanb11) [reply .prettierrc.json]: When I added these to the linter it just broke everything, I fixed the issue with the color tho if you wanted to look through again

File: .prettierrc.json
Comment (harsohailB) [inline .prettierrc.json]: Why was this removed? We need that

File: .eslintrc.json
Comment (harsohailB) [inline .eslintrc.json:3]: I think you removed the indent rule here. We need that as well.

**Review by harsohailB — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (tylerlam1) [inline .gitignore]: Can you add a newline here to get rid of the (-) error?

File: components/Footer.jsx
Comment (harsohailB) [inline components/Footer.jsx]:
```suggestion
        <SocialMediaIcons classes="text-[#7055FD] md:text-white" />
```

nit

File: .gitignore
Comment (harsohailB) [reply .gitignore]: We should add `eol-last: ["error", "always"]` to eslint as well then to avoid this in the future @topanb11

File: components/SocialMediaIcons.jsx
Comment (topanb11) [reply components/SocialMediaIcons.jsx]: Okay I'll take a look at it again tomorrow

File: .eslintrc.json
Comment (topanb11) [reply .eslintrc.json:3]: Added it back

File: .gitignore
Comment (topanb11) [reply .gitignore]: Added it to the eslint file

File: .prettierrc.json
Comment (topanb11) [reply .prettierrc.json]: Added it back

File: components/SocialMediaIcons.jsx
Comment (topanb11) [reply components/SocialMediaIcons.jsx]: Should be good now

File: .eslintrc.json
Comment (harsohailB) [inline .eslintrc.json]: Formatting

**Review by harsohailB — APPROVED** [review]

**Pull request merged**
